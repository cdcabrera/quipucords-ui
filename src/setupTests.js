import { configure, mount, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { act } from 'react-dom/test-utils';
configure({ adapter: new Adapter() });

/**
 * Emulate for component checks
 */
jest.mock('i18next', () => {
  const Test = function () { // eslint-disable-line
    this.use = () => this;
    this.init = () => Promise.resolve();
    this.changeLanguage = () => Promise.resolve();
  };
  return new Test();
});

/**
 * Enzyme for components using hooks.
 *
 * @param {React.ReactNode} component
 * @param {object} options
 * @param {Function} options.callback
 * @param {object} options.options
 *
 * @returns {Promise<null>}
 */
global.mountHookComponent = async (component, { callback, ...options } = {}) => {
  let mountedComponent = null;
  await act(async () => {
    mountedComponent = mount(component, options);
  });
  mountedComponent?.update();

  if (typeof callback === 'function') {
    await act(async () => {
      await callback({ component: mountedComponent });
    });
    mountedComponent?.update();
  }

  return mountedComponent;
};

global.mountHookWrapper = global.mountHookComponent;

/**
 * Enzyme for components using hooks.
 *
 * @param {React.ReactNode} component
 * @param {object} options
 * @param {Function} options.callback
 * @param {object} options.options
 *
 * @returns {Promise<null>}
 */
global.shallowHookComponent = async (component, { callback, ...options } = {}) => {
  let mountedComponent = null;
  await act(async () => {
    mountedComponent = shallow(component, options);
  });
  mountedComponent?.update();

  if (typeof callback === 'function') {
    await act(async () => {
      await callback({ component: mountedComponent });
    });
    mountedComponent?.update();
  }

  return mountedComponent;
};

global.shallowHookWrapper = global.shallowHookComponent;

/*
 * This is a temporary patch for applying a global Jest "beforeAll", based on
 * jest-prop-type-error, https://www.npmjs.com/package/jest-prop-type-error
 */
beforeAll(() => {
  const { error } = console;

  console.error = (message, ...args) => {
    if (/(Invalid prop|Failed prop type)/gi.test(message)) {
      throw new Error(message);
    }

    error.apply(console, [message, ...args]);
  };
});
