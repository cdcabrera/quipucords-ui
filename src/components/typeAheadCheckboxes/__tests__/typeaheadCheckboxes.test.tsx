import React from 'react';
import { shallowComponent } from '../../../../config/jest.setupTests';
import { TypeaheadCheckboxes } from '../typeaheadCheckboxes';

describe('TypeaheadCheckboxes', () => {
  it('should render a basic component', async () => {
    const props = {
      options: [{ value: 'ipsum', label: 'Lorem' }]
    };
    const component = await shallowComponent(<TypeaheadCheckboxes {...props} />);
    expect(component).toMatchSnapshot('basic');
  });
});
