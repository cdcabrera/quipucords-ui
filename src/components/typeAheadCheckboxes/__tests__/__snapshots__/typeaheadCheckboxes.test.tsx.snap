// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeaheadCheckboxes should render a basic component: basic 1`] = `
<Select
  id="multi-typeahead-checkbox-select"
  isOpen={false}
  onOpenChange={[Function]}
  onSelect={[Function]}
  role="menu"
  selected={[]}
  toggle={[Function]}
>
  <SelectList
    id="select-multi-typeahead-checkbox-listbox"
  >
    <SelectOption
      hasCheckbox={true}
      id="select-multi-typeahead-ipsum"
      isFocused={false}
      isSelected={false}
      label="Lorem"
      value="ipsum"
    >
      Lorem
    </SelectOption>
  </SelectList>
</Select>
`;
